src/api/app/models/project.rb

```diff
@@ -1075,23 +1075,22 @@ def repositories_linking_project(tproj)
 
   # called either directly or from delayed job
   def do_project_copy(params)
-    # set user if nil, needed for delayed job in Package model
-    User.session ||= User.find_by!(login: params[:user])
-
-    check_write_access!
+    User.find_by!(login: params[:user]).run_as do
```

Comment:
:thought_balloon: I don't like that this is happening here, it should be in the controller IMHO. I realize that you are just refactoring it...